README.md

```diff
@@ -1 +1,285 @@
+  <ul>
+    <li>Removed <code>order()</code> because its an internal method, reducing the number of conflicts in case you create a class that inherits from <code>ChainedMap</code></li>
+    <li>Removed <code>.clean(obj)</code> because it's more of a helper (it didn't use <code>this.store</code> whatsoever)</li>
+    <li>Added <code>.tap(key, fn)</code></li>
+    <li>Added <code>.keys()</code> just like the native <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/keys">Map</a></li>
+    <li>Added <code>.forEach()</code> just like the native <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach">Map</a></li>
+    <li>Changed <code>.entries()</code> to no longer return <code>undefined</code> if the map is empty in order to avoid any "gotchas"</li>
+  </ul>
+</details>
+
+
+#### .get(key)
+
+Retrieve the value of the entry corresponding to `key`.
+
+#### .set(key, value)
+
+Set the value the entry corresponding to `key`.
+
+Returns itself to allow chaining.
+
+#### .tap(key, fn)
+
+Execute `fn` with the current value of the entry corresponding to `key`.   
+The return value of `fn` will replace the current value.
+
+Returns itself to allow chaining.
+
+#### .delete(key)
+
+Remove entry corresponding to `key`.
+
+Returns itself to allow chaining.
+
+#### .has(key)
+
+Check if there's an entry corresponding to `key`.
+
+Returns `true` if it exists, `false` otherwise.
+
+#### .clear()
+
+Remove all entries from the Map.
+
+Returns itself to allow chaining.
+
+#### .keys()
+
+Returns an array of the keys stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .values()
+
+Returns an array of the values stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .entries()
+
+Returns an object of all the entries stored in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .forEach(fn, [thisArg])
+
+Execute `fn` once per each key/value pair in the Map.
+
+The order will having into consideration entries that called `.before(name)` or `.after(name)`, see [OrderableChainedMap](#orderable-chained-map).
+
+#### .merge(ob, [omit])
+
+Provide an object which maps its properties and values into the backing Map as keys and values.   
+You can also provide an array as the second argument for property names to omit from being merged.
+
+Returns itself to allow chaining.
+
+#### .extend(shorthands)
+
+Adds aliases for each `key` specified in `shorthands`.
+
+Returns itself to allow chaining.
+
+```js
+const chainedMap = new ChainedMap();
+
+chainedMap.extend(['foo', 'bar']);
+chainedMap.foo('foz');
```

Comment:
Should we provide a second argument here to illustrate that these are only setters? Calling these like this would return the ChainedMap, and people might expect these shorthands to also be getters.

Reply:
Will add `.get` examples illustrating that they are setters